.pipelines/templates/with-ios-simulator-steps.yml

```diff
@@ -16,7 +16,7 @@ steps:
 - bash: |
     set -e -x
 
-    if [[ -v ORT_EXTENSIONS_BUILD_SIMULATOR_ID ]]; then
+    if [[ -n "${ORT_EXTENSIONS_BUILD_SIMULATOR_ID}" ]]; then
```

Comment:
try [[ -n ${ORT_EXTENSIONS_BUILD_SIMULATOR_ID-} ]] 
if you want to test if it existing and not empty, then do something.

Reply:
thanks for the tip.
`[[ -n "${ORT_EXTENSIONS_BUILD_SIMULATOR_ID}" ]]` does work in the CI build. I'm not very familiar with bash though, is there a reason to prefer `[[ -n ${ORT_EXTENSIONS_BUILD_SIMULATOR_ID-} ]]`?

Reply:
it is more robust since it can also run with 'bash -u'.

Reply:
Add a letter, for example "x", in front of the string. Then test if the result equals to "x".

Reply:
> Add a letter, for example "x", in front of the string. Then test if the result equals to "x".

I see that that should work too, but why is it better?

Reply:
https://tldp.org/LDP/abs/html/comparison-ops.html#AEN3669 Maybe it is not a concern anymore.